chalk/arrow.py

```diff
@@ -30,27 +30,42 @@ class ArrowOpts:
 def connect(
     self: Diagram, name1: str, name2: str, style: ArrowOpts = ArrowOpts()
 ) -> Diagram:
-    bb1 = self.get_subdiagram_envelope(name1)
-    bb2 = self.get_subdiagram_envelope(name2)
-    return self + arrow_between(bb1.center, bb2.center, style)
+    sub1 = self.get_subdiagram(name1)
```

Comment:
Might consider doing these with withNames? Don't think the code would change much, but it would be a good test.